src/billing_handler.rs

```diff
@@ -2,32 +2,73 @@ use crate::model::AppState;
 
 use actix_web::web::{Data, Json};
 use actix_web::{get, post, HttpResponse, Responder};
-use anyhow::Context;
+use k256::ecdsa::SigningKey;
 use k256::elliptic_curve::generic_array::sequence::Lengthen;
 use serde_json::json;
 use tiny_keccak::{Hasher, Keccak};
 
 #[derive(Debug, serde::Deserialize)]
 pub struct SigningData {
     nonce: String,
-    txhashes: Vec<String>,
+    tx_hashes: Vec<String>,
 }
 
 #[get("/billing/inspect")]
-pub async fn get_bill(appstate: Data<AppState>) -> impl Responder {
+pub async fn inspect_bill(appstate: Data<AppState>) -> impl Responder {
     HttpResponse::Ok().json(json!({
         "bill": appstate.execution_costs.lock().unwrap().clone(),
     }))
 }
 
+#[get("/billing/latest")]
+pub async fn get_last_bill_claim(appstate: Data<AppState>) -> impl Responder {
+    let mut last_bill_claim_guard = appstate.last_bill_claim.lock().unwrap();
+    if last_bill_claim_guard.0.is_none() {
+        return HttpResponse::Ok().body("No bill claimed yet");
```

Comment:
Keep the response schema consistent. This is not a good API design if sending text as body in one case and json in another. Either send error like invalid request (400) upon unavailability of claimed bill. Client will just handle it accordingly. Or semantic can be like interpret empty json fields as claimed bill unavailability.